Upgrade errorprone to 2.19.1

AlexLandau opened the pull request:
## Before this PR
A certain large codebase was blocked from upgrading gradle-baseline because of https://github.com/google/error-prone/issues/3831 (fixed in 2.19.0)

## After this PR
<!-- User-facing outcomes this PR delivers go below -->
==COMMIT_MSG==
Upgrade Error Prone to 2.19.1 (from 2.18.0)
==COMMIT_MSG==

## Possible downsides?
You may want to review the new/modified checks ([2.19.0](https://github.com/google/error-prone/releases/tag/v2.19.0), [2.19.1](https://github.com/google/error-prone/releases/tag/v2.19.1)) for whether checks should be ignored or promoted.

Commit:
Upgrade errorprone to 2.19.1

Commit:
Ignore UnnecessaryTestMethodPrefix

svc-changelog committed:
Add generated changelog entries

Commit:
Fix interaction of using errorProneApply on a disabled check

carterkozak commented:
Was the workaround in #2537 not sufficient to unblock?

AlexLandau commented:
> Was the workaround in #2537 not sufficient to unblock?

It's not the only case where this happens (ImmutableEnumChecker/Immutable is another). I hadn't seen that PR, in any case.

carterkozak commented:
I’m generally in favor of upgrading, and appreciate this PR, but I’ll need to take a close look through the changes and disable any new checks that are likely to impact a lot of repos since that causes baseline to fall behind.

carterkozak commented on gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java:
```diff
@@ -171,12 +173,23 @@ private static void configureErrorProneOptions(
+                    if (specificChecks.isPresent()) {
+                        List<String> errorProneChecks = specificChecks.get();
+                        return Iterables.concat(
+                                errorProneChecks.stream()
```
Could you add an inline comment here describing why this addition is necessary?

AlexLandau replied:
Done

Commit:
Add comment explaining a bug workaround

Commit:
Fix formatting

Commit:
Opt out of potentially disruptive new checks

Note that I haven't opted out of NotJavadoc, but based on rollout
we may follow up to turn off if it's more frictionful than we
expect.

Commit:
Revert "StrictUnusedVariable is not suppressed when UnusedVariable is disabled (#2537)"

This reverts commit db7cd7222bcebf72908e510ecafec77907471c9e.
This reverts the production code change, but retains the new tests which
continue to pass with the new error-prone version.

carterkozak commented:
Sorry for the delay on this one! I've put together a couple minor modifications, could you take a look over them and merge if you're happy with the changes? #2586
Happy to merge once that goes through

AlexLandau commented:
> Sorry for the delay on this one! I've put together a couple minor modifications, could you take a look over them and merge if you're happy with the changes? #2586 Happy to merge once that goes through

I merged your changes in (manually, due to the develop-merge issue you mentioned and because this repo only allows squash merges).

AlexLandau commented:
It looks like the tests are resolving `com.palantir.baseline:baseline-error-prone:latest.release` to `5.11.0`. Meanwhile, running `printVersion` on the PR branch gives `5.7.0-18-gc5fddba.dirty`.

AlexLandau commented:
I gave up on trying to fix the testkit stuff and just rebased.

carterkozak reviewed (approved):
Thanks!

svc-autorelease commented:
Released [5.12.0](https://github.com/palantir/gradle-baseline/releases/tag/5.12.0)
